src/common/datatypes/Path.h

```diff
@@ -79,8 +79,8 @@ struct Step {
   }
 
   bool operator==(const Step& rhs) const {
-    return dst == rhs.dst && type == rhs.type && name == rhs.name && ranking == rhs.ranking &&
-           props == rhs.props;
+    return dst == rhs.dst && (type == rhs.type || type == -rhs.type) && name == rhs.name &&
```

Comment:
Should we compare `dst == src` when type is negative?

Reply:
I think it's not needed. This is comparing two `STEPS`, not `EDGES`, so only the `dst` need to be compared. There is also no `src` member in STEP.

This diff avoids treating two physical versions of the same self-reflective edges as different.